connector_clickup/models/clickup_backend/common.py

```diff
@@ -0,0 +1,364 @@
+                priority=7,
+                with_delay=with_delay,
+                force_update_field=None,
+            )
+
+    @contextmanager
+    def work_on(self, model_name, **kwargs):
+        """Add the work on for clickup."""
+        self.ensure_one()
+        location = self.uri
+        client_id = self.client_id
+        client_secret = self.client_secret
+        token = self.api_key
+        username = self.username
+        password = self.password
+        if self.test_mode:
+            location = self.test_location
+            client_id = self.client_id
+            client_secret = self.client_secret
+            token = self.test_token
+            username = self.username
+            password = self.password
+
+        clickup_location = ClickupLocation(
+            location=location,
+            token=token,
+        )
+
+        # Clickup have different endpoint/credentials for token
+        clickup_location_token = ClickupTokenLocation(
+            location=location,
+            client_id=client_id,
+            client_secret=client_secret,
+            username=username,
+            password=password,
+        )
+        with ClickupAPI(clickup_location, clickup_location_token) as clickup_api:
+            _super = super()
+            # from the components we'll be able to do: self.work.clickup_api
+            with _super.work_on(model_name, clickup_api=clickup_api, **kwargs) as work:
+                yield work
+
+    def export_projects(self, with_delay=True, from_sync=False):
+        for backend in self.sudo():
+            backend._export_from_date(
+                model="clickup.project.project",
+                from_date_field=None if not from_sync else False,
+                with_delay=with_delay,
+            )
+
+    def export_tasks(self, with_delay=True, from_sync=False):
+        for backend in self.sudo():
+            backend._export_from_date(
+                model="clickup.project.tasks",
+                from_date_field=None if not from_sync else False,
+                with_delay=with_delay,
+            )
+
+    def _export_from_date(
+        self,
+        model,
+        from_date_field=None,
+        filters=None,
+        force_update_field=None,
+        priority=None,
+        with_delay=True,
+    ):
+        """Common method for export data from from_date."""
+
+        if not filters:
+            filters = {}
+        import_start_time = datetime.now()
+        job_options = {}
+        if priority or priority == 0:
+            job_options["priority"] = priority
+        for backend in self:
+            from_date = None
+            if from_date_field:
+                from_date = backend[from_date_field]
+                if from_date:
+                    from_date = from_date.strftime("%Y-%m-%dT%H:%M:%SZ")
+                else:
+                    from_date = to_iso_datetime(from_date)
+
+            filters["with_count"] = "true"
```

Comment:
not needed, also force will not used in export, so we can remove the logic of force in exports